### DISCUSSION TITLE
How to dynamically change Layout Animation delay?

### DISCUSSION THREAD

**Discussion by brerdem** [software-mansion/react-native-reanimated#3654] [Q&A]
I'd like to change the a predefined Layout animation's delay based on state value. I tried ```PredefinedAnimation.delay(sharedValue)``` and assigned `sharedValue` based on state but it didn't work.

**Comment by jclif** [comment]
Would also like to accomplish something similar.

**Reply by tomekzaw** [reply]
Hey, I don't think it's possible using predefined animations. However, you can implement a custom layout animation where you can use `sharedValue.value` when calling `withTiming`, `withSpring` etc. You can start by copying the implementation of predefined animation.

**Comment by brerdem** [comment]
@tomekzaw I did it as you suggested. Thanks!

**Reply by jclif** [reply]
I'm not sure I follow the recommendation; would you happen to have a code snippet of what you did to accomplish this?
